### PR TITLE
Interpret reqs as YAML, enable external credits and exclusion in verifier

### PR DESCRIPTION
**Pull request by bnehoran**
- Requirement JSONs are now interpreted as YAML files.
As expected, this passes all tests without any changes needed.
- A few tests are updated to conform to recent changes in the COS and MAE major requirements (from earlier merged pull requests; we just forgot to update the tests back then)
- Minor changes to the verifier to allow counting external credits (AP credits, summer courses, study abroad, transfer credits)
- Enable excluding courses from a req by adding an `excluded_course_list` field

Addresses: #101 (still need frontend support to fully resolve)
Resolves: #173 
Resolves: #100

### CODE REVIEW

File: tigerpath/majors_and_certificates/scripts/verifier.py [hunk 1]
```diff
@@ -530,43 +546,51 @@ def _course_match(course_name, pattern):
     pattern = pattern.split(':')[0] # remove course title
     pattern = ["".join(p.split()).upper() for p in pattern.split('/')] # split by '/' and
     course = ["".join(c.split()).upper() for c in course_name.split('/')] # remove spaces
+    match = False
     for c in course:
         for p in pattern:
+            if p[0] == '~':  # excluded course
```
Comment (churichard) [inline tigerpath/majors_and_certificates/scripts/verifier.py]: Remove support for ~

File: tigerpath/majors_and_certificates/scripts/verifier.py [hunk 2]
```diff
@@ -392,7 +396,8 @@ def _init_path_to(req):
             _init_path_to(subreq)
 
 def _json_format(obj):
-   return json.dumps(obj, sort_keys=False, indent=2, separators=(',', ': ')) + "\n"
+    import json
```
Comment (churichard) [inline tigerpath/majors_and_certificates/scripts/verifier.py:399]: Is there a reason you moved the import here? Might be better to have it at the top of the file

**Review by churichard — CHANGES_REQUESTED** [review]

File: tigerpath/majors_and_certificates/scripts/verifier.py (hunk 2 above)
Comment (bnehoran) [reply tigerpath/majors_and_certificates/scripts/verifier.py:399]: This is just a debugging function and is not called under normal usage, and this is the only place where json is needed.

**Review by churichard — APPROVED** [review]
Looks great!

**Pull request merged**